hilo.rb

```diff
@@ -1,34 +1,121 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+  # Constructor to initialise the class with a maximum number of guesses
+  def initialize(max_guesses)
+    @max_guesses = max_guesses
+    @already_guessed = Array.new
+    @random_num = get_random_num
+  end
 
-    # only allow integer input
+  # The game is started
+  def play
+    welcome
+
+    while game_running?
+      print_guesses_left
+      guess = guess_a_number
+      check_guess(guess)
+    end
+
+    finish_game
+  end    
+
+  private
+
+  def get_random_num
+    random_num = rand(1..100)
+  end
+
+  # Welcome message for the game
+  def welcome
+    puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
+    puts "You will have a total of #{@max_guesses} guesses.\nYou cannot guess the same number more than once."
+    puts ""
+  end
+
+  # Returns true if there are turns left and the number hasn't been guessed
+  def game_running?
+    guesses_left > 0 && !@already_guessed.include?(@random_num)
+  end
+
+  # Returns true if guess equals the random number
+  def has_won?(guess)
+    guess == @random_num
+  end
```

Comment:
You're not using this method anymore

Reply:
Changed to keep this method included, but improve readability in https://github.com/amykingOW/Hi-Lo-Game/pull/4/commits/f17992ddabb1dbe3e7cc364e8bcad1cfed8c1a87.